application/src/main/java/org/opentripplanner/graph_builder/module/DirectTransferGenerator.java

```diff
@@ -115,31 +132,48 @@ public void buildGraph() {
             if (sd.stop.transfersNotAllowed()) {
               continue;
             }
-            distinctTransfers.put(
-              new TransferKey(stop, sd.stop, sd.edges),
-              new PathTransfer(stop, sd.stop, sd.distance, sd.edges)
-            );
+            TransferKey transferKey = new TransferKey(stop, sd.stop, sd.edges);
+            PathTransfer pathTransfer = distinctTransfers.get(transferKey);
+            if (pathTransfer == null) {
+              EnumSet<StreetMode> modes = EnumSet.of(mode);
+              distinctTransfers.put(
+                transferKey,
+                new PathTransfer(stop, sd.stop, sd.distance, sd.edges, modes)
+              );
+            } else {
+              pathTransfer.addMode(mode);
+            }
           }
-          if (OTPFeature.FlexRouting.isOn()) {
-            // This code is for finding transfers from AreaStops to Stops, transfers
-            // from Stops to AreaStops and between Stops are already covered above.
-            for (NearbyStop sd : nearbyStopFinder.findNearbyStops(
-              ts0,
-              transferProfile,
-              transferProfile.journey().transfer(),
-              true
-            )) {
-              // Skip the origin stop, loop transfers are not needed.
-              if (sd.stop == stop) {
-                continue;
-              }
-              if (sd.stop instanceof RegularStop) {
-                continue;
-              }
+        }
+        // Calculate flex transfers if flex routing is enabled.
+        for (RouteRequest transferProfile : flexTransferRequests) {
+          StreetMode mode = transferProfile.journey().transfer().mode();
```

Comment:
Is this just always WALK?

Reply:
I changed this to `StreetMode mode = StreetMode.WALK;`